opencog/openpsi/event.scm

```diff
@@ -0,0 +1,40 @@
+; Todo: Move this to a event.scm file
+; --------------------------------------------------------------
+; Event Detection
+(define psi-event-node (Concept (string-append psi-prefix-str "event")))
+
+(define-public (psi-get-monitored-events)
+"
+  Returns a list containing all the monitored events.
+"
+    (filter
+        (lambda (x) (not (equal? x psi-event-node)))
+        (cog-chase-link 'InheritanceLink 'ConceptNode psi-event-node))
+)
+
+(define (psi-create-monitored-event event)
+	(define event-concept (Concept (string-append psi-prefix-str event)))
+	(Inheritance event-concept psi-event-node)
+	; Initialize value to 0, when an instance of the event occurs, value will be
+	; set to 1.
+	(psi-set-value! event-concept 0)
+	;(format #t "new event: ~a\n" event-concept)
+	event-concept)
+
+
+; --------------------------------------------------------------
+; Event Creation
+(define new-face (psi-create-monitored-event "new-face"))
```

Comment:
The stuff here, and below, all probably belong in `ros-behavior-scripting` or something like that -- not here.